src/main/cpp/src/get_json_object.cu

```diff
@@ -80,106 +60,39 @@ thrust::optional<rmm::device_uvector<path_instruction>> parse_path(
-  {
-    // logically delete child generator
-    output_len += child_generator.get_output_len();
-  }
-
-  CUDF_HOST_DEVICE void write_start_array()
-  {
-    // TODO
-  }
-
-  CUDF_HOST_DEVICE void write_end_array()
-  {
-    // TODO
-  }
-
-  CUDF_HOST_DEVICE void copy_current_structure(json_parser<max_json_nesting_depth>& parser)
-  {
-    // TODO
-  }
-
-  /**
-   * Get current text from JSON parser and then write the text
-   * Note: Because JSON strings contains '\' to do escape,
-   * JSON parser should do unescape to remove '\' and JSON parser
-   * then can not return a pointer and length pair (char *, len),
-   * For number token, JSON parser can return a pair (char *, len)
-   */
-  CUDF_HOST_DEVICE void write_raw(json_parser<max_json_nesting_depth>& parser)
-  {
-    if (output) {
-      auto copied = parser.write_unescaped_text(output + output_len);
-      output_len += copied;
-    }
-  }
-
-  CUDF_HOST_DEVICE inline size_t get_output_len() const { return output_len; }
-
- private:
-  char const* const output;
-  size_t output_len;
-};
-
-/**
- * @brief Result of calling a parse function.
- *
- * The primary use of this is to distinguish between "success" and
- * "success but no data" return cases.  For example, if you are reading the
- * values of an array you might call a parse function in a while loop. You
- * would want to continue doing this until you either encounter an error
- * (parse_result::ERROR) or you get nothing back (parse_result::EMPTY)
- */
-enum class parse_result {
-  ERROR,          // failure
-  SUCCESS,        // success
-  MISSING_FIELD,  // success, but the field is missing
-  EMPTY,          // success, but no data
-};
+template <int max_json_nesting_depth>
+CUDF_HOST_DEVICE bool evaluate_path(json_parser<max_json_nesting_depth>& p,
+                                    json_generator<max_json_nesting_depth>& g,
+                                    write_style style,
+                                    path_instruction const* path_ptr,
+                                    int path_size)
+{
+  return path_evaluator::evaluate_path<max_json_nesting_depth>(
+    p, g, style, path_ptr, path_size);
+}
 
 /**
  * @brief Parse a single json string using the provided command buffer
  *
- * @param j_state The incoming json string and associated parser
- * @param commands The command buffer to be applied to the string. Always ends
- * with a path_operator_type::END
- * @param output Buffer user to store the results of the query
+ * @param j_parser The incoming json string and associated parser
+ * @param path_ptr The command buffer to be applied to the string.
+ * @param path_size Command buffer size
+ * @param output Buffer used to store the results of the query
  * @returns A result code indicating success/fail/empty.
  */
 template <int max_json_nesting_depth = curr_max_json_nesting_depth>
 __device__ parse_result parse_json_path(json_parser<max_json_nesting_depth>& j_parser,
```

Comment:
I see that this function is called only once? If so, we can just eliminate it completely and put the code at the caller.

Reply:
Updated, added annotation: `inline`